### PR TITLE
SendContent

### PR DESCRIPTION
**Pull request by LEGOlord208**
See #351

### CODE REVIEW

**Comment by LEGOlord208** [comment]
One problem I see is that you need to set `MessageEdit: &MessageEdit{}` to be able to use SetContent.

Should I make a getter method? Should I remove the pointer, so it defaults to an empty struct?

**Comment by LEGOlord208** [comment]
Since I accidently changed my local `develop` branch, I used this pull early for one project.

File: message.go [hunk 1]
```diff
@@ -33,18 +33,24 @@ type Message struct {
 
 // MessageSend stores all parameters you can send with ChannelMessageSendComplex.
 type MessageSend struct {
-	Content string        `json:"content"`
-	Tts     bool          `json:"tts"`
-	Embed   *MessageEmbed `json:"embed"`
-	Nonce   string        `json:"nonce"`
+	MessageEdit
+	Tts   bool          `json:"tts"`
+	Embed *MessageEmbed `json:"embed"`
```
Comment (iopred) [inline message.go]: Embed is defined on MessageEdit, you shouldn't need this line.

Comment (LEGOlord208) [inline message.go]: Oo right! Missed that! 1 second!

File: message.go [hunk 2]
```diff
@@ -33,18 +33,24 @@ type Message struct {
...
 }
 
 // MessageEdit stores all parameters you can send with ChannelMessageSendComplex.
 type MessageEdit struct {
```
Comment (iopred) [inline message.go]: Is there a better name here? Do we need the MessageEdit? What if we just document that Tts doesn't work when editing? In that way we could just have 'MessageParameters' as the only struct that both ComplexSend and ComplexEdit take.

Comment (LEGOlord208) [inline message.go]: Right, OK!

**Comment by iopred** [comment]
Lint comment errors breaking the build, but this looks good.

**Comment by LEGOlord208** [comment]
Already on it ;)

File: message.go [hunk 3]
```diff
@@ -31,18 +31,28 @@ type Message struct {
 	Reactions       []*MessageReactions  `json:"reactions"`
 }
 
+// MessageParams is used in MessageSend and MessageEdit to share common parameters.
+type MessageParams struct {
+	Content *string       `json:"content,omitempty"`
+	Embed   *MessageEmbed `json:"embed"`
```
Comment (iopred) [inline message.go]: SORRY! Last thing! This should be omitempty also.

Comment (LEGOlord208) [inline message.go]: Oh, right! No problem!

**Comment by iopred** [comment]
Thanks so much dude! 💯 💯 🥇

**Pull request merged**